packages/widget/src/components/network-selector/network-selector.ts

```diff
@@ -16,70 +17,91 @@ type Direction = (typeof Directions)[keyof typeof Directions];
 export class NetworkSelector extends LitElement {
   static styles = styles;
 
-  @property({
-    type: Boolean
-  })
-  disabled = false;
+  @state()
+  _isDropdownOpen = false;
 
-  @property({
-    type: Boolean
-  })
-  icons = true;
+  @property({ type: Boolean })
+  disabled = false;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   direction?: Direction;
 
-  @property({
-    type: Object
-  })
-  selected?: Domain;
+  @property({ type: Object })
+  selectedNetwork?: Domain;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onNetworkSelected?: (network?: Domain) => void;
 
-  @property({
-    type: Array
-  })
+  @property({ type: Array })
   networks: Domain[] = [];
 
-  onChange({ target }: Event): void {
-    const { value } = target as HTMLOptionElement;
-    const network = this.networks.find((n) => String(n.chainId) == value);
-    this.onNetworkSelected?.(network);
+  _toggleDropdown = (): void => {
+    this._isDropdownOpen = !this._isDropdownOpen;
+  };
+
+  _selectOption(option: Domain, event: Event): void {
+    event.stopPropagation();
+    this.selectedNetwork = option;
+    this.onNetworkSelected?.(option);
+    this._isDropdownOpen = false;
```

Comment:
probably better to use this before callback is called. Never know how long callback execution is gonna take 🤷🏻‍♂️

Reply:
Yeah, it's a good idea 👍